dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavResult.java

```diff
@@ -0,0 +1,374 @@
+        if (children != null) {
+            ArrayList<NavResult> list = new ArrayList<NavResult>();
+            for (NavResult nn : children) {
+                if (nn.isFolder()) {
+                    // for folders we avoid returning the same instance
+                    // it could be changed elsewhere and we need it to
+                    // load its children lazily
+                    NavResult ff = new NavResult(folderId, nn.hostId, nn.folderId, nn.languageId);
+                    ff.setTitle(nn.getTitle());
+                    ff.setHref(nn.getHref());
+                    ff.setOrder(nn.getOrder());
+                    ff.setType(nn.getType());
+                    ff.setPermissionId(nn.getPermissionId());
+                    ff.setShowOnMenu(nn.isShowOnMenu());
+                    list.add(ff);
+                } else {
+                    list.add(nn);
+                }
+            }
+            return list;
+        } else {
+            return new ArrayList<NavResult>();
+        }
+    }
+
+    public String getParentPath() throws DotDataException, DotSecurityException {
+        if (parent == null)
+            return null; // no parent! I'm the root folder
+        if (parent.equals(FolderAPI.SYSTEM_FOLDER))
+            return "/";
+        Folder folder = APILocator.getFolderAPI()
+            .find(parent, sysuser, true);
+        Identifier ident = APILocator.getIdentifierAPI()
+            .find(folder);
+        return ident.getURI();
+    }
+
+    public NavResult getParent() throws DotDataException, DotSecurityException {
+        String path = getParentPath();
+        if (path != null) {
+            NavResult lazyMe = new NavTool().getNav(APILocator.getHostAPI()
+                .find(hostId, sysuser, true), path, languageId, sysuser);
+
+            return lazyMe;
+        } else
+            return null;
+    }
+
+    public String toString() {
+        if (!isCodeLink()) {
+            String titleToShow;
+            try {
+                titleToShow = getTitle();
+            } catch (Exception e) {
+                titleToShow = title;
+            }
+            return "<a href='" + getHref() + "' title='" + titleToShow + "'>" + titleToShow + "</a>";
+        } else {
+            return getCodeLink();
+        }
+    }
+
+    public List<String> getChildrenFolderIds() {
+        return childrenFolderIds != null ? childrenFolderIds : new ArrayList<String>();
+    }
+
+    public void setChildrenFolderIds(List<String> childrenFolderIds) {
+        this.childrenFolderIds = childrenFolderIds;
+    }
+
+    @Override
+    public Iterator<NavResult> iterator() {
+        try {
+            List<? extends NavResult>  childs= getChildren();
+            List<NavResult> ret = new ArrayList<>();
+            for(NavResult child  : childs) {
+                ret.add(child);
+            }
+            return ret.iterator();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)